apps/naive/lib/naive/strategy.ex

```diff
@@ -76,6 +84,27 @@ defmodule Naive.Strategy do
           settings
         )
 
+      :update_buy_position ->
+        current_buy_order = update_order_position(position, :buy_order, trade_event.order_status)
+
+        generate_decisions(
+          rest,
+          [{:skip, %{position | buy_order: current_buy_order}}] ++ generated_results,
+          trade_event,
+          settings
+        )
+
+      :update_sell_position ->
+        current_sell_order =
+          update_order_position(position, :sell_order, trade_event.order_status)
+
+        generate_decisions(
+          rest,
+          [{:skip, %{position | sell_order: current_sell_order}}] ++ generated_results,
+          trade_event,
+          settings
+        )
```

Comment:
Com a introdução de order events algumas variáveis não encaixaram muito bem o nome, como esse `trade_event` que na verdade agora corresponde a um `order_event` (destacado abaixo). Podemos ou separar esses caminhos em funções que aceitam `trade_event` e funções que aceitam `order_event` ou renomear simplesmente para `event`. Talvez a primeira opção seja mais fácil de seguir a lógica, mas acho que não vamos conseguir escapar de alguma duplicação de código. O que acham @NeoVier  @lucca65?
```suggestion
      :update_buy_position ->
        current_buy_order = update_order_position(position, :buy_order, order_event.order_status)

        generate_decisions(
          rest,
          [{:skip, %{position | buy_order: current_buy_order}}] ++ generated_results,
          trade_event,
          settings
        )

      :update_sell_position ->
        current_sell_order =
          update_order_position(position, :sell_order, order_event.order_status)

        generate_decisions(
          rest,
          [{:skip, %{position | sell_order: current_sell_order}}] ++ generated_results,
          order_event,
          settings
        )
```

Reply:
Talvez podemos ter um conjunto de funções que servem como base (que recebem um `event`), e criar outras funções mais especializadas (que recebem um `trade_event` ou `order_event`) que usam as funções base e adicionam o que for necessário, talvez operando em uma estrutura de dados genérica, que pode ser criada a partir de qualquer uma das estruturas mais específicas.

De qualquer forma, acho que as duas opções (receber só `event` ou duplicar um pouco de código) são válidas. Nessa fase mais exploratória, talvez valha a pena manter tudo como `order_event`, e podemos revisitar isso mais tarde

Reply:
Também acho que é algo que podemos ver decidir mais para frente, quando tivermos um escopo mais global do projeto definido. Até porque talvez tenhamos que reecrever isso posteriormente de qualquer jeito